src/Workspaces/MSBuild/Core/Rpc/RemoteBuildHost.cs

```diff
@@ -33,6 +33,16 @@ public async Task<RemoteProjectFile> LoadProjectFileAsync(string projectFilePath
         return new RemoteProjectFile(_client, remoteProjectFileTargetObject);
     }
 
+    /// <summary>Permits loading a project file which only exists in-memory, for example, for file-based program scenarios.</summary>
+    /// <param name="projectFilePath">A path to a project file which may or may not exist on disk. Note that an extension that is known by MSBuild, such as .csproj or .vbproj, should be used here.</param>
```

Comment:
Excellent warning to add! :smile:

Reply:
...except this should go on the interface since that's what's actually seen by consumers.

Reply:
I think this is a good place to document it. This is the API which the project system will call.